Sources/cte/Parser.swift

```diff
@@ -159,6 +159,66 @@ struct Parser {
             let stmtIf = AstNode.If(condition: cond, thenStmt: body, elseStmt: elseStmt)
             return AstNode(stmtIf, tokens: [ifToken, elseToken])
 
+        case .keywordFor:
+            let forToken = advance()
+
+            // Check for infinite loop
+            guard lexer.peek()?.kind != .lbrace else {
+                let body = expression()
+                let stmtFor = AstNode.For(initialiser: nil, condition: nil, step: nil, body: body)
+                return AstNode(stmtFor, tokens: [forToken])
+            }
+
+            var exprs: [AstNode] =  []
+            while exprs.count <= 3 {
+                let token = lexer.peek()?.kind
+                guard token != .semicolon else {
+                    let semicolon = advance()
+                    exprs.append(AstNode(AstNode.Empty(), tokens: [semicolon]))
+                    continue
+                }
+
+                guard token != .lbrace else {
+                    // Check if we have 2 cases already. If so, that means the
+                    // 3rd case is empty. If not, we unexpectedly hit an `{`
+                    if exprs.count == 2 {
```

Comment:
`if exprs.count != 1` would be clearer since it is the only valid case when you hit a `lbrace` prior to it having 3 statements.